src/main/java/com/intuit/graphql/orchestrator/apollofederation/EntityExtensionContext.java

```diff
@@ -0,0 +1,38 @@
+package com.intuit.graphql.orchestrator.apollofederation;
+
+import com.intuit.graphql.graphQL.FieldDefinition;
+import com.intuit.graphql.graphQL.TypeDefinition;
+import com.intuit.graphql.orchestrator.schema.ServiceMetadata;
+import graphql.schema.FieldCoordinates;
+import lombok.Builder;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+@Builder
+@Getter
+public class EntityExtensionContext  {
+
+  public static final String DELIMITER = ":";
+
+  private FieldDefinition fieldDefinition;
+  private TypeDefinition parentTypeDefinition;
+  private boolean requiresTypeNameInjection;
+  private ServiceMetadata serviceMetadata;
+  //private EntityDefinition baseType;
+  private EntityExtensionDefinition thisEntityExtensionDefinition;
+
+  public String createDataLoaderKey() {
```

Comment:
precalculate and save it, change it to get so you don't calculate more than once

Reply:
yup. thanks.  calculating only once  [here](https://github.com/intuit/graphql-orchestrator-java/pull/47/files#diff-7f95dacb01235e440261ab955dbba43c5cbac06056fb2256bcc486da7fd43b75R64)